umm-spec-lib/src/cmr/umm_spec/umm_g/measured_parameters.clj

```diff
@@ -0,0 +1,101 @@
+(ns cmr.umm-spec.umm-g.measured-parameters
+  "Contains functions for parsing and generating the UMM-G dialect for measured parameters."
+  (:require
+   [cmr.umm.umm-granule :as g])
+  (:import cmr.umm.umm_granule.UmmGranule))
+
+(defn- umm-g-qa-stats->QAStats
+  "Returns umm-lib QAStats from given UMM-G QAStats."
+  [qa-stats]
+  (when qa-stats
+    (let [{:keys [QAPercentMissingData
+                  QAPercentOutOfBoundsData
+                  QAPercentInterpolatedData
+                  QAPercentCloudCover]} qa-stats]
+      (g/map->QAStats
+       {:qa-percent-missing-data QAPercentMissingData
+        :qa-percent-out-of-bounds-data QAPercentOutOfBoundsData
+        :qa-percent-interpolated-data QAPercentInterpolatedData
+        :qa-percent-cloud-cover QAPercentCloudCover}))))
+
+(defn- QAStats->umm-g-qa-stats
+  "Returns UMM-G QAStats from given umm-lib QAStats."
+  [qa-stats]
+  (when qa-stats
+    (let [{:keys [qa-percent-missing-data
+                  qa-percent-out-of-bounds-data
+                  qa-percent-interpolated-data
+                  qa-percent-cloud-cover]}
+          qa-stats]
+      {:QAPercentMissingData qa-percent-missing-data
+       :QAPercentOutOfBoundsData qa-percent-out-of-bounds-data
+       :QAPercentInterpolatedData qa-percent-interpolated-data
+       :QAPercentCloudCover qa-percent-cloud-cover})))
+
+(defn- umm-g-qa-flags->QAFlags
+  "Returns UMM-G QAFlags from given umm-lib QAFlags."
+  [qa-flags]
+  (when qa-flags
+    (let [{:keys [AutomaticQualityFlag
+                  AutomaticQualityFlagExplanation
+                  OperationalQualityFlag
+                  OperationalQualityFlagExplanation
+                  ScienceQualityFlag
+                  ScienceQualityFlagExplanation]}
+
+          qa-flags]
+      (g/map->QAFlags
+       {:automatic-quality-flag AutomaticQualityFlag
+        :automatic-quality-flag-explanation AutomaticQualityFlagExplanation
+        :operational-quality-flag OperationalQualityFlag
+        :operational-quality-flag-explanation OperationalQualityFlagExplanation
+        :science-quality-flag ScienceQualityFlag
+        :science-quality-flag-explanation ScienceQualityFlagExplanation}))))
+
+(defn- QAFlags->umm-g-qa-flags
+  "Returns umm-lib QAFlags from given UMM-G QAFlags."
+  [qa-flags]
+  (when qa-flags
+    (let [{:keys [automatic-quality-flag
+                  automatic-quality-flag-explanation
+                  operational-quality-flag
+                  operational-quality-flag-explanation
+                  science-quality-flag
+                  science-quality-flag-explanation]}
+          qa-flags]
+      {:AutomaticQualityFlag automatic-quality-flag
```

Comment:
We need to make sure the umm-lib quality flags are converted into valid UMM-G quality flags, i.e. a default ENUM value needs to be picked for umm-lib quality flags that are not in the UMM-G ENUM list; otherwise we will get invalid UMM-G records during conversion.

Reply:
Which one.

Reply:
I meant which value heh.

Reply:
Erich should be able to provide the default values.

Reply:
Yes I know, I talked to him yesterday, and you as well so I'm not sure what the purpose of this comment is.

Reply:
I have added "Undetermined" as a new enumeration value to all three: AutomaticQualityFlag, OperationalQualityFlag and ScienceQualityFlag, in the https://git.earthdata.nasa.gov/projects/EMFD/repos/unified-metadata-model/browse/granule/v1.4/umm-g-json-schema.json file.  If the current UMM-G model has nil for an element then DON'T assign the new UMM-G v1.4 a value - leave it nil, they are optional.  If the current UMM-G model has a different value then what exists in the new UMM-G v 1.4 model, then use the Undetermined value.  Please also append to accompanying explanation element the following only if the current UMM-G quality flag has a value: "The original records value of {current UMM-G model quality flag value} was not recognized and the value of Undetermined was assigned." That way when analyzers of the data using UMM-G v1.4 can help the data providers clean up their records. This shouldn't happen though because only ECS DAACs use these elements and the new UMM-G v1.4 model is using the same enumeration names as the BMGT (Bulk Metadata Generation Tool).